decidim-forms/app/helpers/decidim/forms/admin/application_helper.rb

```diff
@@ -33,6 +33,17 @@ def dynamic_title(title, **options)
             truncate translated_attribute(title), length: options[:max_length], omission: options[:omission]
           end
         end
+
+        def edit_questionnaire_title
+          case questionnaire.questionnaire_for.class.name
+          when "Decidim::Templates::Template"
+            t(:title, scope: "decidim.forms.admin.questionnaires.form", questionnaire_for: translated_attribute(@template.name))
+          when "Decidim::Surveys::Survey"
+            t(:title, scope: "decidim.forms.admin.questionnaires.form", questionnaire_for: translated_attribute(current_component.name))
+          else
+            t(:title, scope: "decidim.forms.admin.questionnaires.form", questionnaire_for: translated_attribute(questionnaire_for.try(:title)))
+          end
```

Comment:
This creates a dependecy from `decidm-forms` to `decidim-templates` and `decidim-surveys`.
Is it really necessary to have a different default title?
Some alternatives can be:
- to pass the `questionnaire_for` as a parameter
- to move the parts of this method to each of the modules that are using it via override or some other way

Reply:
@tramuntanal good catch, I've refactored it so the title can be overriden in the controller.